src/api/procedures/inviteExternalAgent.ts

```diff
@@ -127,7 +111,7 @@ export async function prepareInviteExternalAgent(
       type: AuthorizationType.BecomeAgent,
       value: permissions,
     };
-    rawAuthorizationData = authorizationDataResolver(permissions, context);
+    rawAuthorizationData = authorizationToAuthorizationData(postTransactionAuthorization, context);
```

Comment:
This is almost perfect, but we're still repeating most of

```
{
  type: AuthorizationType.BecomeAgent,
  value: permissions,
};
```

My idea was having something like 

```
const createBecomeAgentData = (value: KnownPermissionGroup | CustomPermissionGroup) => ({
  type: AuthorizationType.BecomeAgent,
  value,
});
```

And then use that in both branches

Reply:
Makes sense, I made the change.